File: tarefa008/Estado.ts
import { Pais } from "./Pais";

export class Estado {
  constructor(nome:string, sigla:string) {
    this.nome = nome;
    this._sigla = sigla;
  }

  //atributos
  public nome:string;
  private _sigla:string;
  private _pais:Pais;

  //setters
  set pais(novoPais:Pais) {
    this.pais = novoPais;
  }

  //getters
  public get sigla() {
    return this._sigla;
  }

  public get pais() {
    return this._pais;
  }
}

File: tarefa008/java/Estado.java
package tarefa008.java;

/**
 * Classe que representa a entidade Estado |
 * Autor: João Máiro |
 * Data: 15/07/22
 */
public class Estado {
  Estado(String nome, String sigla) {
    this.nome = nome;
    this.sigla = sigla;
  }

  //atributos
  private String nome;
  private String sigla;
  private Pais pais;

  /**
   * Método que modifica o valor do atributo pais
   * @param novoPais
   */
  public void setPais(Pais novoPais) {
    this.pais = novoPais;
  }

  /**
   * Método que retorna o valor do atributo nome
   * @return
   */
  public String getNome() {
    return this.nome;
  }

  /**
   * Método que retorna o valor do atributo sigla
   * @return
   */
  public String getSigla() {
    return this.sigla;
  }

  /**
   * Método que retorna o valor do atributo pais
   * @return
   */  
  public Pais getPais() {
    return this.pais;
  }
}
